core/jvm/src/main/scala/cats/effect/unsafe/WorkerThread.scala

```diff
@@ -415,6 +415,23 @@ private final class WorkerThread(
 
       ((state & ExternalQueueTicksMask): @switch) match {
         case 0 =>
+          if (pool.blockedThreadDetectionEnabled && pool.workerThreads.length > 1) {
+            // TODO prefetch pool.workerThread or Thread.State.BLOCKED ?
+            // TODO check that branch elimination makes it free when off
+            var otherIdx = random.nextInt(pool.workerThreads.length)
+            while (otherIdx == idx) {
+              otherIdx = random.nextInt(pool.workerThreads.length)
```

Comment:
Rather than looping, can we go with a more constant-time approach here? Something like:

```suggestion
              otherIdx = (idx + Math.max(1, random.nextInt(pool.workerThreads.length - 1))) % pool.workerThreads.length
```

This would guarantee that `otherIdx` always ends up *not* on `idx` on our second try, so long as there is more than one thread.

Reply:
Nice! Is that sufficiently more expensive that we don't want to do this on the first attempt?

Reply:
I just worry about pathological cases that are hard to track down. I'd rather be constant time if at all possible. If we don't want to do the fancy modular arithmetic thing, IMO the next-best thing would be to just turn the `while` into an `if` above and, if we happen to randomly select our own index, just bail out and get it next time.